src/contrib/cluster/Akka.Cluster.Sharding/ShardedDaemonProcess.cs

```diff
@@ -150,29 +203,71 @@ public void Init(string name, int numberOfInstances, Func<int, Props> propsFacto
                 TimeSpan.Zero, // passivation disabled
                 StateStoreMode.DData,
                 shardingBaseSettings.TuningParameters,
-                shardingBaseSettings.CoordinatorSingletonSettings, 
+                shardingBaseSettings.CoordinatorSingletonSettings,
                 shardingBaseSettings.LeaseSettings);
 
-            if (string.IsNullOrEmpty(shardingSettings.Role) || Cluster.Get(_system).SelfRoles.Contains(shardingSettings.Role))
+            if (string.IsNullOrEmpty(shardingSettings.Role) ||
+                Cluster.Get(_system).SelfRoles.Contains(shardingSettings.Role))
             {
                 var sharding = ClusterSharding.Get(_system);
                 var shardingRef = sharding.Start(
-                    typeName: $"sharded-daemon-process-{name}",
+                    typeName: FormatWorkerProcessName(name),
                     entityPropsFactory: entityId => propsFactory(int.Parse(entityId)),
                     settings: shardingSettings,
                     messageExtractor: new MessageExtractor(numberOfShards),
-                    allocationStrategy: sharding.DefaultShardAllocationStrategy(shardingSettings), 
+                    allocationStrategy: sharding.DefaultShardAllocationStrategy(shardingSettings),
                     stopMessage ?? PoisonPill.Instance);
 
                 _system.ActorOf(
                     KeepAlivePinger.Props(settings, name, entityIds, shardingRef),
                     $"ShardedDaemonProcessKeepAlive-{name}");
+                
+                return _system.ActorOf(
+                    Props.Create(() =>
+                        new DaemonMessageRouter(entityIds,
+                            shardingRef)), $"ShardedDaemonProcessRouter-{name}");
             }
+            
+            _system.Log.Warning(
+                "ShardedDaemonProcess [{0}] not started because the role [{1}] is not included in the self roles [{2}]",
+                name, shardingSettings.Role, string.Join(", ", Cluster.Get(_system).SelfRoles));
+
+            return null;
+        }
+        
+        private static string FormatWorkerProcessName(string name) => $"sharded-daemon-process-{name}";
+
+        /// <summary>
+        /// Starts a proxy for a sharded daemon process running in a different role.
+        /// </summary>
+        /// <param name="name">The name of this daemon worker set</param>
+        /// <param name="numberOfInstances">The number of instances that belong on this set -
+        /// note that this value must match the value used in the `Init` call on the hosting-nodes
+        /// or messages may be lost.</param>
+        /// <param name="role">The role where the worker actors are hosted.</param>
+        /// <returns>A reference to a router actor that will distribute all messages evenly across the workers
+        /// using round-robin message routing.</returns>
+        public IActorRef InitProxy(string name, int numberOfInstances, string role)
```

Comment:
This is new - since we added push-based communication to interact with `ShardedDaemonProcess`es I thought it also made sense to leverage `ShardRegionProxy` to allow non-hosting nodes to communicate with the worker processes. The only issue here is that if the user gets the number of entities wrong, then our routing logic will be skewed or broken. Not too different than if the user gets the message extractor, role, or entity type wrong so it's not a big deal IMHO.